tableau-server-single-node/azuredeploy.json

```diff
@@ -159,18 +166,18 @@
     "subnet1Name": "Subnet-1",
     "subnet1Prefix": "10.0.0.0/24",
     "numberOfInstances": 1,
-    "ubuntuscripturi": "[uri(parameters('_artifactsLocation'), concat('scripts/config-linux.sh', parameters('_artifactsLocationSasToken')))]",
+    "linuxscripturi": "[uri(parameters('_artifactsLocation'), concat('scripts/config-linux.sh', parameters('_artifactsLocationSasToken')))]",
     "installscripturi": "[uri(parameters('_artifactsLocation'), concat('scripts/automated-installer', parameters('_artifactsLocationSasToken')))]",
     "virtualMachineSize": "[parameters('VMSize')]",
     "imageReference": {
-      "publisher": "Canonical",
-      "offer": "UbuntuServer",
-      "sku": "16.04.0-LTS",
-      "version": "latest"
+      "publisher": "[if(equals(parameters('OS'), 'RHEL'), 'RedHat', if(equals(parameters('OS'), 'Ubuntu'),'Canonical', 'OpenLogic'))]",
```

Comment:
remove the if() statements and use a object variable index - see: https://github.com/bmoore-msft/AzureRM-Samples/blob/master/ContosoPortal/ContosoPortal/modules/vm-module.json#L219-L238